src/commands/trace/trace.ts

```diff
@@ -0,0 +1,127 @@
+      ],
+    ],
+  })
+  private command?: string[]
+
+  private config = {
+    apiKey: process.env.DATADOG_API_KEY || process.env.DD_API_KEY,
+    envVarTags: process.env.DD_TAGS,
+  }
+
+  private name?: string
+
+  public async execute() {
+    if (!this.command || !this.command.length) {
+      this.context.stderr.write('Missing command to run\n')
+
+      return 1
+    }
+
+    const [command, ...args] = this.command
+    const id = crypto.randomBytes(5).toString('hex')
+    const startTime = new Date().toISOString()
+    const spawnResult = spawnSync(command, args, {env: {...process.env, DD_CUSTOM_PARENT_ID: id}, stdio: 'inherit'})
+    const endTime = new Date().toISOString()
+    const exitCode = spawnResult.status ?? this.signalToNumber(spawnResult.signal) ?? 127
+    const api = this.getApiHelper()
+    const [provider, data] = this.getData()
+    const commandStr = this.command.join(' ')
+    await api.reportCustomSpan(
+      {
+        command: commandStr,
+        custom: {
+          id,
+          parent_id: process.env.DD_CUSTOM_PARENT_ID,
+        },
+        data,
+        end_time: endTime,
+        is_error: exitCode !== 0,
+        name: this.name ?? commandStr,
+        start_time: startTime,
+        tags: this.config.envVarTags ? parseTags(this.config.envVarTags.split(',')) : {},
+      },
+      provider
+    )
+
+    return exitCode
+  }
+
+  public getData(): [string, Record<string, string>] {
+    if (process.env.CIRCLECI) {
+      return [
+        'circleci',
+        this.getEnvironmentVars([
+          'CIRCLE_BRANCH',
+          'CIRCLE_BUILD_NUM',
+          'CIRCLE_BUILD_URL',
+          'CIRCLE_JOB',
+          'CIRCLE_NODE_INDEX',
+          'CIRCLE_NODE_TOTAL',
+          'CIRCLE_PROJECT_REPONAME',
+          'CIRCLE_PULL_REQUEST',
+          'CIRCLE_REPOSITORY_URL',
+          'CIRCLE_SHA1',
+          'CIRCLE_TAG',
+          'CIRCLE_WORKFLOW_ID',
+        ]),
+      ]
+    }
+    throw new Error('Cannot detect any CI Provider. This command only works if run as part of your CI.')
+  }
+
+  private getApiHelper(): APIHelper {
+    if (!this.config.apiKey) {
+      this.context.stdout.write(
+        `Neither ${chalk.red.bold('DATADOG_API_KEY')} nor ${chalk.red.bold('DD_API_KEY')} is in your environment.\n`
+      )
+      throw new Error('API key is missing')
+    }
+
+    return apiConstructor(this.getBaseIntakeUrl(), this.config.apiKey)
+  }
+
+  private getBaseIntakeUrl() {
+    if (process.env.DATADOG_SITE || process.env.DD_SITE) {
+      return `https://webhooks-http-intake.logs.${process.env.DATADOG_SITE || process.env.DD_SITE}`
+    }
+
+    return 'https://webhooks-http-intake.logs.datadoghq.com'
+  }
+
+  private getEnvironmentVars(keys: string[]): Record<string, string> {
+    return keys.filter((key) => key in process.env).reduce((accum, key) => ({...accum, [key]: process.env[key]!}), {})
+  }
+
+  private signalToNumber(signal: string | null): number | undefined {
+    if (!signal) {
```

Comment:
FYI: an empty string `''` will also map to `undefined`. I don't know if that's an issue

Reply:
`signal` should never be an empty string, according to the documentation. If it turns out to be an empty string, I think it is OK to map to undefined